qemu/tests/hugepage_reset.py

```diff
@@ -0,0 +1,76 @@
+import time
+import logging
+
+from virttest import env_process
+from virttest import test_setup
+from virttest import utils_test
+from virttest import utils_misc
+from virttest import error_context
+from virttest.staging import utils_memory
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Re-assign nr-hugepages
+
+    1. Set up hugepage with 1G page size and hugepages=8
+    2. Boot up guest using /mnt/kvm_hugepage as backend in QEMU CML
+    3. Change the nr_hugepages after the guest boot up (6 and 10)
+    4. Run the stress test **inside guest**
+    5. change the nr_hugepages after the stress test (6 and 10)
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+
+    def set_hugepage():
+        """Set nr_hugepages"""
+        for h_size in (origin_nr - 2, origin_nr + 2):
+            hp_config.target_hugepages = h_size
+            hp_config.set_hugepages()
+            if params.get('on_numa_node'):
+                logging.info('Set hugepage size %s to target node %s' % (
+                    h_size, target_node))
+                hp_config.set_node_num_huge_pages(h_size, target_node,
+                                                  hugepage_size)
+
+    origin_nr = int(params['origin_nr'])
+    host_numa_node = utils_misc.NumaInfo()
+    mem = int(float(utils_misc.normalize_data_size("%sM" % params["mem"])))
+    if params.get('on_numa_node'):
+        for target_node in host_numa_node.get_online_nodes_withmem():
+            node_mem_free = host_numa_node.read_from_node_meminfo(
+                target_node, 'MemFree')
+            if node_mem_free > mem:
```

Comment:
03:31:24 ERROR| TypeError: '>' not supported between instances of 'str' and 'int'

Reply:
@huangyum Sorry I did not test it after update, having no proper host on hand, still need bother you help me test the updated one, thanks a lot.